### DISCUSSION TITLE
ForecasterAutoregMultiSeries predicts current datetime instead of last history hour

### DISCUSSION THREAD

**Discussion by noafaran1000** [JoaquinAmatRodrigo/skforecast#782] [General]
def train(series_dict, exog_dict, pivot_df):
    end_train = '2024-06-30 23:59:00'
    series_dict_train = {k: v.loc[: end_train, ] for k, v in series_dict.items()}
    exog_dict_train = {k: v.loc[: end_train, ] for k, v in exog_dict.items()}
    series_dict_test = {k: v.loc[end_train:, ] for k, v in series_dict.items()}
    exog_dict_test = {k: v.loc[end_train:, ] for k, v in exog_dict.items()}

    pivot_df.index = pd.to_datetime(pivot_df.index)
    pivot_df = pivot_df.asfreq('H')
    filtered_df = pivot_df[pivot_df.index > pd.to_datetime(end_train)]

    regressor = HistGradientBoostingRegressor(
        max_iter=103,
        max_depth=9,
        learning_rate=0.09669050850160849,
        random_state=123
    )
    best_forecaster = ForecasterAutoregMultiSeries(
        regressor=regressor,
        lags=[1, 2, 3, 4, 5, 6, 7],
        transformer_series=StandardScaler(),
        encoding='ordinal',
        dropna_from_series=False
    )

    best_forecaster.fit(series=series_dict_train, exog=exog_dict_train, suppress_warnings=True)
    predictions = best_forecaster.predict(steps=5, last_window=filtered_df, exog=exog_dict_test, suppress_warnings=True)
    print(predictions)
    return best_forecaster
    
    This is my code, and I dont know why the prediction is for the 5 hours from now. I want it to be predicted 5 hours from the last sample in the last_window,
  
What am i doing wrong?
Thankes.

**Comment by JoaquinAmatRodrigo** [comment]
Hi @noafaran1000 

Could you add at the end of your code this two lines and show the results?
```python
print(filtered_df.tail())
print(predictions.head())
```
